src/main/scala/barstools/iocell/chisel/IOCell.scala

```diff
@@ -59,7 +59,9 @@ class DigitalInIOCellBundle extends Bundle {
   val ie = Input(Bool())
 }
 
-trait IOCell extends BaseModule
+trait IOCell extends BaseModule {
+  var i_name: String
```

Comment:
Can you make this name more descriptive? Also try
```
var iocell_name: Option[String] = None
override def suggestName(s: String): IOCell = {
  iocell_name = Some(s)
  super.suggestName(s)
}
```

If you do it this way, you don't have to set a default in the classes which inherit this trait. 
Also you don't have to do another suggestName in generateIOCell

Reply:
Yeah. It minimizes the amount of code which has to change, any thing which tries to name an IOCell with suggestName will also set the stored io_cell name with this change